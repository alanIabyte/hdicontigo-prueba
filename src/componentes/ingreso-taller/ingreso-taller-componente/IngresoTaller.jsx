import React from "react";
import PropTypes from "prop-types";
import {
  EnvolvedorIngresoTaller,
  IngresoTallerCampo,
  IngresoTallerValor,
} from "./IngresoTaller.styled";
import obtenerValorDeArregloDeStrings from "../../utilidades-compartidas/obtenerValorDeArregloDeStrings";
import agregarFormatoDeFecha from "../../utilidades-compartidas/agregarFormatoFecha";

const diccionario = {
  fechaIngreso: "Fecha de ingreso",
};

const IngresoTaller = (props) => {
  const { datosIngresoTaller } = props;

  const fechaIngreso = obtenerValorDeArregloDeStrings(
    datosIngresoTaller,
    "Fechaingreso: "
  );

  const fechaIngresoFormato = agregarFormatoDeFecha(fechaIngreso);

  return (
    <EnvolvedorIngresoTaller>
      <IngresoTallerCampo>{diccionario.fechaIngreso}</IngresoTallerCampo>
      <IngresoTallerValor>{fechaIngresoFormato}</IngresoTallerValor>
    </EnvolvedorIngresoTaller>
  );
};

IngresoTaller.defaultProps = {
  datosIngresoTaller: ["Fechaingreso:  "],
};

IngresoTaller.propTypes = {
  datosIngresoTaller: PropTypes.arrayOf(PropTypes.string),
};

export default IngresoTaller;
