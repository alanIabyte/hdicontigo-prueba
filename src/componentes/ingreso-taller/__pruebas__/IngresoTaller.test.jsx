import React from "react";
import renderer from "react-test-renderer";
import IngresoTaller from "../ingreso-taller-componente/IngresoTaller";

describe("IngresoTaller", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<IngresoTaller />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
