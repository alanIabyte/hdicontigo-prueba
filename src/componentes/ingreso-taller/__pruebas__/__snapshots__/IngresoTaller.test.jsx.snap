// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngresoTaller se despliega correctamente 1`] = `
<div
  className="sc-ikkxIA"
>
  <div
    className="sc-kpDqfm sc-dAbbOL bTCkFB hdZOgq"
  >
    Fecha de ingreso
  </div>
  <div
    className="sc-kpDqfm sc-feUZmu bTCkFB jMhLTa"
  >
    Fecha inv√°lida
  </div>
</div>
`;
